server/service/jct/api.coffee

```diff
@@ -787,51 +783,54 @@ API.service.jct.permission = (issn, institution) ->
+  if res_sa and res_sa.compliant is 'no'
+    return res_sa
+
+  # Get rights retention data
+  res_r = API.service.jct.retention journal, undefined
+  # merge the qualifications and logs from SA prohibition
+  res_r = _merge_logs_and_qa(res_sa, res_r)
+  if res_r.compliant is 'yes'
+    return res_r
 
   # Get OA.Works permission
   rs = API.service.jct.permission journal, institution
 
-  # merge the qualifications and logs from SA prohibition into OA.Works permission
-  rs.qualifications ?= []
-  if res_sa.qualifications? and res_sa.qualifications.length
-    for q in (if _.isArray(res_sa.qualifications) then res_sa.qualifications else [res_sa.qualifications])
-      rs.qualifications.push(q)
-  rs.log ?= []
-  if res_sa.log? and res_sa.log.length
-    for l in (if _.isArray(res_sa.log) then res_sa.log else [res_sa.log])
-      rs.log.push(l)
+  # merge the qualifications and logs from SA prohibition and SA rights retention into OA.Works permission
+  rs = _merge_logs_and_qa(res_r, rs)
 
   # check for retention
   if rs
     _rtn = {}
     for r in (if _.isArray(rs) then rs else [rs])
-      if r.compliant isnt 'yes' and retention
+      if r.compliant isnt 'yes'
         # only check retention if the funder allows it - and only if there IS a funder
         # funder allows if their rights retention date
         if journal and funder? and fndr = API.service.jct.funders funder
           r.funder = funder
-          # 1609459200000 = Wed Sep 25 52971
-          # if fndr.retentionAt? and (fndr.retentionAt is 1609459200000 or fndr.retentionAt >= Date.now())
-          # if retentionAt is in past - active - https://github.com/antleaf/jct-project/issues/437
           if fndr.retentionAt? and fndr.retentionAt < Date.now()
+            r.compliant = 'yes'
             r.log.push code: 'SA.FunderRRActive'
-            # 26032021, as per https://github.com/antleaf/jct-project/issues/380
-            # rights retention qualification disabled
-            #r.qualifications ?= []
-            #r.qualifications.push 'rights_retention_funder_implementation': funder: funder, date: moment(fndr.retentionAt).format 'YYYY-MM-DD'
-            # retention is a special case on permissions, done this way so can be easily disabled for testing
-            _rtn[journal] ?= API.service.jct.retention journal
-            r.compliant = _rtn[journal].compliant
-            r.log.push(lg) for lg in _rtn[journal].log
-            if _rtn[journal].qualifications? and _rtn[journal].qualifications.length
-              r.qualifications ?= []
-              r.qualifications.push(ql) for ql in _rtn[journal].qualifications
+            # 22022022, as per https://github.com/antleaf/jct-project/issues/503
+            # Add rights retention author qualification
+            r.qualifications ?= []
+            r.qualifications.push({'rights_retention_author_advice': ''})
           else
             r.log.push code: 'SA.FunderRRNotActive'
```

Comment:
After SA.FunderRRNotActive is triggered, the next stage in the SA workflow says to add SA.Unknown or SA.NotCompliant based on what the result of the OAB check was.  Perhaps this is being done elsewhere using that merge code above, but I can't see it easily.

Reply:
Implemented this